components/tidb_query_datatype/src/codec/row/v2/encoder_for_test.rs

```diff
@@ -89,8 +90,100 @@ impl Column {
     }
 }
 
+/// Checksum
+/// - HEADER(1 byte)
+///   - VER: version(3 bit)
+///   - E:   has extra checksum
+/// - CHECKSUM(4 bytes)
+///   - little-endian CRC32(IEEE) when hdr.ver = 0 (default)
+pub trait CheckSumHandler {
+    // update_col updates the checksum with the encoded value of the column.
+    fn checksum(&mut self, buf: &[u8]) -> Result<()>;
+
+    // header_value returns the checksum header value.
+    fn header_value(&self) -> u8;
+
+    // value returns the checksum value.
+    fn value(&self) -> u32;
+}
+
+pub struct Crc32RowChecksumHandler {
+    header: ChecksumHeader,
+    hasher: crc32fast::Hasher,
+}
+
+impl CheckSumHandler for Crc32RowChecksumHandler {
+    fn checksum(&mut self, buf: &[u8]) -> Result<()> {
+        self.hasher.update(buf);
+        Ok(())
+    }
+
+    fn header_value(&self) -> u8 {
+        self.header.value()
+    }
+
+    fn value(&self) -> u32 {
+        self.hasher.clone().finalize()
+    }
+}
+
+pub struct ChecksumHeader(u8);
+
+impl ChecksumHeader {
+    fn new() -> Self {
+        ChecksumHeader(0)
+    }
+
+    #[cfg(test)]
+    fn set_version(&mut self, ver: u8) {
+        self.0 = ver & 0b0000_0111;
```

Comment:
Is it better to use `self.0 |= ...`, otherwise `set_version` will revert other bit fields of the header.